lib/serviceDiscovery/engines/consul/index.js

```diff
@@ -116,13 +124,19 @@ class ConsulService extends Service {
 			this._updateNetwork(data || []);
 		});
 
+		this._watch.on('error', (error) => {
+			logger.error.data(error).log('error while running discovery process');
+		});
+
 		logger.debug('discovery started');
 	}
 
 	/**
 	 * Stop discovery and announcements
 	 */
 	close() {
+		this._announces = [];
```

Comment:
If people wanna abuse private members, they do so at their own risk. We shouldn't have to jump through hoops just in case, imho.